mindmeld/models/text_models.py

```diff
@@ -29,33 +29,31 @@
 from sklearn.svm import SVC
 from sklearn.tree import DecisionTreeClassifier
 
+from .evaluation import EvaluatedExample, StandardModelEvaluation
 from .helpers import (
     CHAR_NGRAM_FREQ_RSC,
     QUERY_FREQ_RSC,
     WORD_FREQ_RSC,
     WORD_NGRAM_FREQ_RSC,
-    register_model,
 )
-from .model import EvaluatedExample, Model, StandardModelEvaluation
+from .model import ModelConfig, Model, PytorchModel
 
-_NEG_INF = -1e10
-
-# classifier types
-LOG_REG_TYPE = "logreg"
-DECISION_TREE_TYPE = "dtree"
-RANDOM_FOREST_TYPE = "rforest"
-SVM_TYPE = "svm"
-SUPER_LEARNER_TYPE = "super-learner"
-BASE_MODEL_TYPES = [LOG_REG_TYPE, DECISION_TREE_TYPE, RANDOM_FOREST_TYPE, SVM_TYPE]
+logger = logging.getLogger(__name__)
 
-# default model scoring type
-ACCURACY_SCORING = "accuracy"
 
+class TextModel(Model):
```

Comment:
Unclear why we need to inherit from Model. These fields can be represented as Enums